File: java/src/Leetcode643.java
public class Leetcode643 {
    public double findMaxAverage(int[] nums, int k) {
        int largest = Integer.MIN_VALUE;
        int curr = 0;
        for(int i = 0; i < k; i++){
            curr += nums[i];
        }
        largest = curr;
        for(int i = k; i < nums.length; i++){
            curr = curr + nums[i] - nums[i - k];
            largest = Math.max(curr, largest);
        }

        return largest * (1.0) / k;
    }
}

File: python/Leetcode643.py
from typing import List

class Leetcode643:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        len_nums = len(nums)
        ans_sum = curr = sum(nums[0:k])
        for i in range(k, len_nums):
            curr = curr + nums[i] - nums[i-k]
            ans_sum = max(ans_sum, curr)
        return ans_sum / k